prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -287,21 +287,28 @@ def get_storage_controller_data(self) -> Tuple[Dict[str, int], Dict[str, List]]:
             for storage_id in storage_ids:
                 # eg: /redfish/v1/Systems/1/Storage/XYZ123
                 curr_storage_uri = (
-                    RedfishHelper._storage_root_uri(system_id, storage_name) + storage_id
+                    RedfishHelper._storage_root_uri(system_id, storage_name) + "/" + storage_id
                 )
 
                 # list of storage controllers for that storage id
-                storage_controllers_list: List[Dict] = self.redfish_obj.get(curr_storage_uri).dict[
-                    "StorageControllers"
-                ]
-                storage_controller_count[system_id] += len(storage_controllers_list)
+                storage_controllers = self.redfish_obj.get(curr_storage_uri).dict
+                if "StorageControllers" in storage_controllers:
+                    storage_controllers_list: List[Dict] = storage_controllers.get("StorageControllers")
+                if "Controllers" in storage_controllers:
```

Comment:
With this approach we can end up with `storage_controllers_list` undefined. We should raise some exception in with else.